### PR TITLE
Enable setting menu to be accessible via NTP

### PR DESCRIPTION
**Pull request by imptrx**
Spec: https://github.com/brave/brave-browser/issues/5041

## Changes
- You can now open the settings menu via keyboard

Quick View:
keyboard-settings
## Test plan
- Go to new tabs page 
- Press tab 
- There should be an outline when you reach the settings icon at the bottom left of the page
- Pressing `space` or `enter` will open the setting menu
- Pressing tab will allow you to select the setting menu icon again
- Pressing `space` or `enter` will open the setting menu

##### Link / storybook path to visual changes
- https://brave-ui-fgdae6j9x.now.sh/?path=/story/feature-components-new-tab-default--page

## Integration
- [ ] Does this contain changes to src/components or src/
  - [ ] Will you publish to npm immediately after this PR, or wait until sometime in the future?
  - [ ] Incompatible API change to something existing _(major version increase)_
  - [ ] Adding new backwards-compatible functionality? _(minor version increase)_
  - [ ] Fixing a bug backwards-compatibly? _(patch version increase)_
  
- [ ] Does this contain changes to src/features for brave-core?
  - [x] Are there non backwards-compatible changes required for brave-core? **Do not merge until brave-core PR is approvable.** Link to brave-core PR: 
- https://github.com/brave/brave-core/pull/2814
  - [x] Will you create brave-core PR to update to this commit after it is merged?
  - [ ] Wants uplift to brave-core feature branch?
     - When uplift-approved, merge to brave-core-0.VV.x feature branch
     - Create additional brave-core PRs for each feature branch to update commit

### CODE REVIEW

File: stories/features/newTab/default/footerInfo.tsx [hunk 1]
```diff
@@ -21,6 +21,13 @@ interface Props {
 }
 
 export default class FooterInfo extends React.PureComponent<Props, {}> {
+
+  onKeyPressSettings = (e: React.KeyboardEvent<HTMLButtonElement>) => {
```
Comment (petemill) [inline stories/features/newTab/default/footerInfo.tsx]: Why is this necessary? If this is a <button element then we get this for free, along with other accessibility benefits.

Comment (imptrx) [reply stories/features/newTab/default/footerInfo.tsx]: This handler was added in order to add a conditional to the onKeyDown for which button was pressed. Otherwise, simple pressing "Tab" through the page element will toggle the menu. 

We wanted to implement the functionality to "click outside the menu to close it" which caused a double trigger on the button when we clicked on it. Thus, the reason I had to add an additional set of logic around `onKeyDown` was because `onClick` had to be disabled in that case.

**Comment by imptrx** [comment]
Closing this PR as all NTP has been moved to brave-core. Will include all changes here in https://github.com/brave/brave-core/pull/2814
